### PR TITLE
[CELEBORN-1801][FOLLOWUP] Extract RemoteShuffleEnvironment, NettyShuffleEnvironmentWrapper, SimpleResultPartitionAdapter to flink common module

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

Extract `RemoteShuffleEnvironment`, `NettyShuffleEnvironmentWrapper`, `SimpleResultPartitionAdapter` to flink common module. Meanwhile, `RemoteShuffleInputGate` and `RemoteShuffleResultPartition` are abstracted in flink common module.

### Why are the changes needed?

After removing out-of-dated flink 1.14 and 1.15 in #3029, `RemoteShuffleEnvironment`, `NettyShuffleEnvironmentWrapper`, `SimpleResultPartitionAdapter` could be extracted to flink common module. Meanwhile, `RemoteShuffleInputGate` and `RemoteShuffleResultPartition` could also be abstracted in flink common module.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

CI.

### CODE REVIEW

File: client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/SimpleResultPartitionAdapter.java
Comment (reswqa) [inline client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/SimpleResultPartitionAdapter.java:1]: I wonder if this adapter is still needed?

Comment (SteNicholas) [reply client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/SimpleResultPartitionAdapter.java:1]: @reswqa, yep. The `SimpleResultPartitionAdapter` isn't needed for `RemoteShuffleMaster`.

**Review by reswqa — APPROVED** [review]
Nice refactor!

**Comment by reswqa** [comment]
Merged to main(v0.6.0)

**Pull request closed without merging**